include/deal.II/physics/elasticity/kinematics.h

```diff
@@ -0,0 +1,367 @@
+       * The differential operator $\nabla_{0}$ is defined as
+       * $\frac{\partial}{\partial \mathbf{X}}$.
+       */
+      template <int dim, typename Number>
+      SymmetricTensor<2, dim, Number>
+      e (const Tensor<2, dim, Number> &Grad_u);
+
+      /**
+       * Returns the Almansi strain SymmetricTensor,
+       * as constructed from the deformation gradient tensor @p F.
+       * The result is expressed as
+       * @f[
+       *  \mathbf{A} := \frac{1}{2} \left[ \mathbf{I}
+       *   - \mathbf{F}^{-T}\cdot\mathbf{F}^{-1} \right] \, .
+       * @f]
+       */
+      template <int dim, typename Number>
+      SymmetricTensor<2, dim, Number>
+      a (const Tensor<2, dim, Number> &F);
+
+//@}
+
+      /**
+       * @name Strain rate tensors
+       */
+//@{
+
+      /**
+       * Returns the spatial velocity gradient Tensor,
+       * as constructed from the deformation gradient tensor @p F
+       * and its material time derivative @p dF_dt (the material velocity
+       * gradient).
+       * The result is expressed as
+       * @f[
+       *  \mathbf{l} := \dot{\mathbf{F}}\cdot\mathbf{F}^{-1} \, .
+       * @f]
+       */
+      template <int dim, typename Number>
+      Tensor<2, dim, Number>
+      l (const Tensor<2, dim, Number> &F,
+         const Tensor<2, dim, Number> &dF_dt);
+
+      /**
+       * Returns the rate of deformation Tensor (also known as the rate of strain
+       * Tensor), as constructed from the deformation gradient tensor @p F
+       * and its material time derivative @p dF_dt (the material velocity
+       * gradient).
+       * The result is expressed as
+       * @f[
+       *  \mathbf{d} := \frac{1}{2} \left[ \mathbf{l} + \mathbf{l}^{T} \right] \, .
+       * @f]
+       * where $\mathbf{l}$ is the spatial velocity gradient Tensor.
+       */
+      template <int dim, typename Number>
+      SymmetricTensor<2, dim, Number>
+      d (const Tensor<2, dim, Number> &F,
+         const Tensor<2, dim, Number> &dF_dt);
+
+      /**
+       * Returns the rate of rotation Tensor (also known as the vorticity
+       * Tensor), as constructed from the deformation gradient tensor @p F
+       * and its material time derivative @p dF_dt (the material velocity
+       * gradient).
+       * The result is expressed as
+       * @f[
+       *  \mathbf{w} := \frac{1}{2} \left[ \mathbf{l} - \mathbf{l}^{T} \right] \, .
+       * @f]
+       * where $\mathbf{l}$ is the spatial velocity gradient Tensor.
+       */
+      template <int dim, typename Number>
+      Tensor<2, dim, Number>
+      w (const Tensor<2, dim, Number> &F,
+         const Tensor<2, dim, Number> &dF_dt);
+
+//@}
+    };
+
+  }
+}
+
+
+
+#ifndef DOXYGEN
+// ------------------------- inline functions ------------------------
+
+
+
+template <int dim, typename Number>
+inline
+Tensor<2, dim, Number>
+Physics::Elasticity::Kinematics::F (const Tensor<2, dim, Number> &Grad_u)
+{
+// See Wriggers (2008) p 23 equ. 3.14
```

Comment:
There are two ways to define such macros. For the first one, take a look at `doc/doxygen/options.dox.in`, around line 82 and following: These commands just define macros with one argument that will simply be expanded to something else, with no further postprocessing of arguments. In your case, you want two arguments -- for this, take a look at https://www.stack.nl/~dimitri/doxygen/manual/custcmd.html .

We have a second way where we can do some actual processing of the arguments. I don't think you'll actually need this -- you just have page and equation numbers, nothing needs to happen to them -- but that can be done in a filter script that we run before doxygen even sees anything. That would be in `doc/doxygen/scripts/filter, and examples start at line 82. (What a funny coincidence that it's the same line number as above :-) )